### ISSUE TITLE
Add an analyzer to suggest non-preferred overloads

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3188]
Originated from https://github.com/dotnet/roslyn-analyzers/pull/3074#discussion_r353764913

> Having put more thought in this, I think the TryParse case above falls into the more generic analysis:
> 
>- I have `n` different overloads for a method for similar functionality which were added over a time period.
> 
>- I would prefer to have an analyzer that allows me to flag use of non-preferred overloads
> 
>- Analyzer would have some default set of BCL APIs where we already know the preferred and non-preferred overloads. In our case, this will be the `TryParse` methods.
> 
>- Analyzer would have an option to allow me to specify fully qualified signatures of preferred (and/or non-preferred) overloads for custom APIs.
>
> The above seems to very closely match the [BannedApiAnalyzers](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md) [](/dotnet/roslyn-analyzers/blob/HEAD@{2019-12-04T14:11:36Z}/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md), though that analyzer does not special case banned APIs from BCL libraries and is a much more stricter version of _banned_ as opposed to _non-preferred_ APIs. I am wondering if we should add a new rule to that analyzer with a different ID and much less stricter message, that only deals with preferred and non-preferred symbol pairs, special cases the APIs that we already know about (TryParse group) and allows users to specify an additional customized preferences in `BannedSymbols.txt`. Building on the example from [here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md) [](/dotnet/roslyn-analyzers/blob/HEAD@{2019-12-04T14:11:36Z}/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md):
> 
> ```csharp
> class MyType
> {
>    void Method(T1 t);
>    void Method(T2 t);  // preferred overload
> }
> ```
> 
> Potential "BannedSymbols.txt":
> 
> ```
> MyType.Method(T1);prefer:MyType.Method(T2)
> ```

**Comment by Evangelink** [comment]
This new analyzer could also be extended to support types, this way we could suggest generic types over classical ones as we do in rule CA1010 (see #1490)
